### PR TITLE
Resizers overhaul and dejank

### PR DESCRIPTION
**Pull request by andycreeth**

### CODE REVIEW

File: app/components/LiveDock.vue.ts [hunk 1]
```diff
@@ -196,6 +188,13 @@ export default class LiveDock extends Vue {
     this.chatService.refreshChat();
   }
 
+  get hideChat() {
+    // previewEnabled is poorly named.  It really just means stuff
```
Comment (blackxored) [inline app/components/LiveDock.vue.ts]: Nitpick: use block comments

File: app/components/pages/Live.vue [hunk 2]
```diff
@@ -5,65 +5,49 @@
...
+      <div class="live-preview-container" :style="{ flex: '0 1 ' + displayWidth + 'px' }">
+        <div class="studio-controls-top">
+          <h4 class="studio-controls__label">
+            {{ $t('Preview') }}
+          </h4>
+          <div v-if="!performanceModeEnabled">
+            <i
+              v-if="previewEnabled"
+              class="icon-view icon-button icon-button--lg"
+              @click="previewEnabled = false"
```
Comment (blackxored) [inline app/components/pages/Live.vue]: Would prefer a method for this, not having this kind of logic in templates.

File: app/components/pages/Live.vue [hunk 3]
```diff
@@ -5,65 +5,49 @@
...
+              v-tooltip="disablePreviewTooltip"/>
+            <i
+              v-if="!previewEnabled"
+              class="icon-hide icon-button icon-button--lg"
+              @click="previewEnabled = true"
```
Comment (blackxored) [inline app/components/pages/Live.vue]: See the previous comment.

File: app/components/pages/Live.vue [hunk 4]
```diff
@@ -137,57 +86,35 @@
...
 
 .live-display-wrapper {
-  width: 100%;
-}
-
-.live-display {
-  width: 100%;
-  height: 100%;
+  max-width: 100%;
+  .radius();
```
Comment (blackxored) [inline app/components/pages/Live.vue]: Nitpick: put mixins above other rules.

File: app/components/pages/Live.vue.ts [hunk 5]
```diff
@@ -92,8 +83,12 @@ export default class Live extends Vue {
     this.customizationService.setSettings({ bottomdockSize: value });
   }
 
+  get displayWidth() {
+    return (16 / 9) * (this.height - 29);
+  }
+
   get maxHeight() {
-    return this.$root.$el.getBoundingClientRect().height;
+    return this.$root.$el.getBoundingClientRect().height - 400;
```
Comment (blackxored) [inline app/components/pages/Live.vue.ts]: Magic number?

File: app/components/windows/Main.vue.ts [hunk 6]
```diff
@@ -98,10 +106,15 @@ export default class Main extends Vue {
     return this.userService.isLoggedIn();
   }
 
-  mainContentsRight = false;
```
Comment (blackxored) [inline app/components/windows/Main.vue.ts]: Nice one!

File: app/components/windows/Main.vue.ts [hunk 7]
```diff
@@ -98,10 +106,15 @@ export default class Main extends Vue {
...
+  get renderDock() {
+    return this.isLoggedIn && !this.isOnboarding && this.hasLiveDock;
+  }
+
+  get dockCollapsed() {
```
Comment (blackxored) [inline app/components/windows/Main.vue.ts]: Nitpick: would prefer `is` or `has` prefixes for boolean-returning functions.

File: app/components/LiveDock.vue.ts (hunk 1 above)
Comment (andycreeth) [reply app/components/LiveDock.vue.ts]: I find normal comments easier than block comments.  Block comments take up more space and are annoying.

File: app/components/pages/Live.vue (hunk 4 above)
Comment (andycreeth) [reply app/components/pages/Live.vue]: Is this an actual thing people do?  Why?

**Pull request merged**